[Dataset quality] Rename malformed docs to degraded docs

yngrdyn opened the pull request:
### Before
<img width="2411" alt="image" src="https://github.com/elastic/kibana/assets/1313018/8336c098-eaa2-4ea1-b111-bd0c00c6a4d7">

### After
<img width="2410" alt="image" src="https://github.com/elastic/kibana/assets/1313018/12d67d55-4c32-4a6c-bc39-afb9267ec18a">

yngrdyn committed:
Renamed column

apmmachine commented:
### :robot: GitHub comments

<details><summary>Expand to view the GitHub comments</summary>
<p>

Just comment with:
- `/oblt-deploy` : Deploy a Kibana instance using the Observability test environments.
- `/oblt-deploy-serverless` : Deploy a serverless Kibana instance using the Observability test environments.
- `run` `elasticsearch-ci/docs` : Re-trigger the docs validation. (use unformatted text in the comment!)

</p>
</details>

mdbirnstiehl commented:
@yngrdyn degraded documents works for me. One note/nitpick it should be em dashes with no spaces (—) instead of hyphens (-) around the "documents with the _ignored property"

yngrdyn commented:
@mdbirnstiehl even thought is not so visual in the code changes 😅  I already removed the hyphens and place the em dashes. Updated the `After` screenshot for reference

felixbarny commented:
It doesn't need to be within this PR, but we should also rename other occurrences of "malformed documents" to "degraded documents". For example: `malformedDocsColumnName`, `GET /internal/dataset_quality/data_streams/malformed_doc`.

yngrdyn commented:
@felixbarny oh, you meant the full rename of it, not only in the UI 😅 
I will addressed in this one then

felixbarny commented:
Yeah, I think we should be consistent in how we refer to it internally, including the code base, as well as how we present it to users.

I suppose it's fine to do that after the FF, though. So no rush.

yngrdyn committed:
Merge branch 'main' into rename-malformed-docs-column

tonyghiani reviewed (approved):
LGTM!

yngrdyn committed:
Renamed route from malformed_docs to degraded_docs

yngrdyn committed:
showing default for degraded docs

yngrdyn committed:
Merge branch 'main' into rename-malformed-docs-column

elasticmachine commented:
Pinging @elastic/apm-ui (Team:APM)

cauemarcondes commented on packages/kbn-apm-synthtrace/src/scenarios/degraded_logs.ts:
```diff
@@ -30,10 +30,10 @@ const scenario: Scenario<LogDocument> = async (runOptions) => {
       // "ignore_above": 1024 in mapping
-      const MALFORMED_LOG_LEVEL = MORE_THAN_1024_CHARS;
+      const DEGRADED_LOG_LEVEL = MORE_THAN_1024_CHARS;
 
       // "ignore_above": 1024 in mapping
-      const MALFORMED_CLOUD_REGION = MORE_THAN_1024_CHARS;
+      const DEGRADED_CLOUD_REGION = MORE_THAN_1024_CHARS;
```
Can't you use `MORE_THAN_1024_CHARS` and not create these two extra variables?

yngrdyn replied:
Addressed in [55814ec](https://github.com/elastic/kibana/pull/172586/commits/55814ec91335a449ed4a99829e76909d2f500dc0).

cauemarcondes commented on packages/kbn-apm-synthtrace/src/scenarios/degraded_logs.ts:
```diff
@@ -76,7 +76,7 @@ const scenario: Scenario<LogDocument> = async (runOptions) => {
           .dataset('synth.2')
           .message(MESSAGE_LOG_LEVELS[index].message as string)
-          .logLevel(isMalformed ? MALFORMED_LOG_LEVEL : MESSAGE_LOG_LEVELS[index].level)
+          .logLevel(isMalformed ? DEGRADED_LOG_LEVEL : MESSAGE_LOG_LEVELS[index].level)
```
Like:
```suggestion
          .logLevel(isMalformed ? MORE_THAN_1024_CHARS : MESSAGE_LOG_LEVELS[index].level)
```

yngrdyn committed:
CR changes: use MORE_THAN_1024_CHARS directly

cauemarcondes reviewed (approved):
LGTM

yngrdyn committed:
Merge branch 'main' into rename-malformed-docs-column

elasticmachine commented:
Pinging @elastic/obs-ux-logs-team (Team:obs-ux-logs)
